department-of-veterans-affairs/caseflow
# TYLERB/APPEALS-9150: Standardize Task Action Modal Button Text

Resolves [APPEALS-9150](https://vajira.max.gov/browse/APPEALS-9150) and [APPEALS-9281](https://vajira.max.gov/browse/APPEALS-9281)

### Description
Standardized task action modal submit button text across all pre-docket modals. Where possible, it was changed to pull the text from the task action repository.

### Acceptance Criteria
- [x] CAMO Modals
  - [x] Assign to Program Office (Assign)
  - [x] Send to Board Intake (Send)
- [x] VHA PO Modals
  - [x]  Put task on hold (Put task on hold)
  - [x] End hold early (End hold)
  - [x] Ready for review (Send)
  - [x] Assign to VISN (Assign)
  - [x] Return to CAMO (Return)
  - [x] Mark task in progress (Mark in progress)
- [x] EMO Modals 
  - [x] Assign to RPO (Assign)  
  - [x] Return to Board Intake (Return)
  - [x] Ready for review (Send)
- [x] Education RPO Modals 
  - [x] Return to Executive Management Office (Return)
  - [x] Ready for review (Send)
  - [x] Mark task in progress (Mark in progress)
- [x] BVA Intake Modals
  - [x] Docket Appeal (Confirm??) 
  - [x] Return appeal to VHA CAMO (Return)
  - [x] Return appeal to Education Service (Return)

### Testing Plan
1. Setup some appeals and tasks to be used for testing
```ruby
5.times do
  appeal = FactoryBot.create(:appeal, :with_pre_docket_task)
  parent_task = appeal.tasks.last
  emo_document_search_task=EducationDocumentSearchTask.create!( parent: parent_task, appeal: appeal, assigned_at: Time.zone.now, assigned_to: EducationEmo.singleton )
end

5.times do
  appeal = FactoryBot.create(:appeal, :with_pre_docket_task)
  parent_task = appeal.tasks.last
  vha_document_search_task=VhaDocumentSearchTask.create!( parent: parent_task, appeal: appeal, assigned_at: Time.zone.now, assigned_to: VhaCamo.singleton )
end
```

2. Swap to a CAMO user.   There is one with the cssid of CAMOUSER. Navigate to the Camo organization queue.

3. Select one of the created appeals and go to the details page.

4. Select the Send to Board Intake task action and verify the button text is "Send"

5. Send one of the appeals back to intake for later testing

6. Select the Assign to Program Office task action and verify the button text is "Assign"

7. Send a task/appeal to a program office of your choosing.

8. Switch to a program office user VHAPOUSER is the css id and navigate to the program office queue that you sent the appeal to

9. Select that appeal and go to the details page

10. Select the Ready for Review task action and verify the button text is "Send"

11. Select the Assign to VISN task action and verify the button text is "Assign"

12. Select the Return to Camo team task action and verify the button text is "Return"

13. Select the Mark task in progress task action and verify the button text is "Mark in progress"

14. Select the Put task on hold task action and verify the button text is "Put task on hold"

15. Place the task on hold

16. Select the End hold early task action and verify the button text is "End hold"
 
17. Switch to an emo user the css id is EMOUSER
18. Select an appeal and go to the details page
19. Select the Assign to regional processing office task action and verify the button text is "Assign"
20. Send an appeal to an office of your choosing for po testing and select a new appeal
21. Select the Return to board intake task action and verify the button text is "Return"
22. Send a task back to intake for later testing and select a new appeal
23. Select the Ready for review task action and verify the button text is "Send"
24. Swap to an edu rpo user the css id is EDURPOUSER 
25. Go to the program office queue you sent the earlier appeal to
26. Select the appeal and go to the case details page
27. Select the Return to Executive Management Office task action and verify the button text is "Return"
28. Select the Ready for Review task action and verify the the button text is "Send"
29. Select the mark task in progress task action and verify the button text is "Mark in progress"
30. Swap to a BVA intake user css id is BVAISHAW
31. Navigate to the queue, go to the review tab and select the appeal returned from Camo
32. Select the Docket Appeal task action and verify the button text is "Confirm" ? unsure about this one
33. Select the Return appeal to VHA task action and verify the button text is "Return"
34. Navigate back to the queue, go to the review tab, and select the appeal returned from Emo
35. Select the Return appeal to Education Service task action and verify the button text is "Return"
